internal.nix

```diff
@@ -62,27 +62,74 @@ rec {
   # Type: Path -> Derivation
   patchedLockfile = file: writeText "packages-lock.json" (builtins.toJSON (patchLockfile file));
 
+  # Prepared source for npm installation
+  nodeSource = nodes: runCommand "node-sources-${nodejs.version}"
+    { } ''
+    tar --no-same-owner --no-same-permissions -xf ${nodejs.src}
+    mv node-* $out
+  '';
+
   node_modules =
     { src
     , packageJson ? src + "/package.json"
     , packageLockJson ? src + "/package-lock.json"
     , buildInputs ? [ ]
     , nativeBuildInputs ? [ ]
     , nodejs ? default_nodejs
+    , preBuild ? ""
+    , postBuild ? ""
+    , preInstallLinks ? { } # set that describes which files should be linked in a specific packages folder
```

Comment:
This needs better documentation and a test. Right now it is only being tested wit the tweag-www example that I posted in the comments.

Reply:
A bit more docs have been added and a test has also been added.